packages/sources/coinpaprika/src/endpoint/crypto.ts

```diff
@@ -27,38 +59,102 @@ export const inputParameters: InputParameters = {
+  // We'll reset the response data to not output the entire CP coins list
+  const result = Requester.withResult({ ...response, data: {} }, undefined, payload)
+  return Requester.success(jobRunID, result, true, batchablePropertyPath)
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
   const validator = new Validator(request, inputParameters)
   if (validator.error) throw validator.error
 
   const jobRunID = validator.validated.id
-  const symbol = validator.overrideSymbol(AdapterName) as string
-  const quote = validator.validated.data.quote
+  const symbol = validator.overrideSymbol(AdapterName)
+  const requestedQuotes = validator.validated.data.quote
   const coinid = validator.validated.data.coinid as string | undefined
 
-  // If coinid was provided or base was overridden, that symbol will be fetched
-  let coin = coinid || (symbol !== validator.validated.data.base && symbol)
-  if (!coin) {
-    const coinIds = await getCoinIds(context, jobRunID)
-    coin = await getSymbolToId(symbol, coinIds)
-  }
+  const url = 'v1/tickers'
+  const resultPath = validator.validated.data.resultPath || endpointResultPaths.crypto
 
-  const url = `v1/tickers/${coin.toLowerCase()}`
-  const resultPath = validator.validated.data.resultPath
-
-  const params = {
-    quotes: quote.toUpperCase(),
+  let quotes: string
+  if (Array.isArray(requestedQuotes)) {
+    quotes = requestedQuotes.map((quote) => quote.toUpperCase()).join(',')
+  } else {
+    quotes = requestedQuotes.toUpperCase()
   }
 
+  const params = { quotes }
   const options = {
     ...config.api,
     url,
     params,
   }
 
-  const response = await Requester.request(options)
-  response.data.result = Requester.validateResultNumber(response.data, resultPath)
-  response.data.cost = 2
+  if (Array.isArray(symbol)) {
+    const requestedData: RequestedData[] = []
+
+    for (let i = 0; i < symbol.length; i++) {
+      if (symbol[i] !== validator.validated.data.base[i]) {
+        requestedData.push({ coinid: symbol[i] })
+      } else {
+        requestedData.push({ symbol: symbol[i] })
+      }
+    }
+
+    const response = await Requester.request<ResponseSchema[]>(options)
+    return handleBatchedRequest(jobRunID, request, response, requestedData, resultPath)
+  }
+
+  // If coinid was provided or base was overridden, that symbol will be fetched
+  const coin = coinid || (symbol !== validator.validated.data.base && symbol ? symbol : undefined)
+
+  const response = await Requester.request<ResponseSchema[]>(options)
+
+  const coinData = getCoin(response.data, symbol, coin)
+  if (!coinData) {
+    throw new Error(`unable to find coin: ${coin || symbol}`)
+  }
 
-  return Requester.success(jobRunID, response, config.verbose)
+  const result = Requester.validateResultNumber(coinData, [
+    'quotes',
+    requestedQuotes.toUpperCase(),
+    resultPath,
+  ])
+  return Requester.success(jobRunID, Requester.withResult(response, result), config.verbose)
```

Comment:
```suggestion
  return Requester.success(jobRunID, Requester.withResult(response, result), config.verbose, batchablePropertyPath)
```

Also need `batchablePropertyPath` on unbatched requests so that it knows they should be part of the batch warmer.